Server/test/userFunctions.ts

```diff
@@ -0,0 +1,269 @@
+/* eslint-disable node/no-unpublished-import */
+import request from 'supertest';
+import expect from 'expect';
+import {UserTest} from '../src/interfaces/User';
+import ErrorResponse from '../src/interfaces/ErrorResponse';
+import LoginMessageResponse from '../src/interfaces/LoginMessageResponse';
+import randomstring from 'randomstring';
+
+const getUsers = (url: string | Function): Promise<UserTest[]> => {
```

Comment:
getUsers function does not currently handle errors that may occur during the GraphQL query. Adding appropriate error handling would be a good improvement..

for example :

`const getUsers = (url: string | Function): Promise<UserTest[]> => {
  return new Promise((resolve, reject) => {
    request(url)
      .post('/graphql')
      .set('Content-Type', 'application/json')
      .send({query: '{ users { id user_name email } }'})
      .expect(200)
      .then(response => {
        const users = response.body.data.users;
        if (!users || !Array.isArray(users)) {
          _// If the response doesn't contain the expected user array, handle the error._
          reject(new Error('Invalid response format'));
        } else {
          _// Valid response, resolve with the users._
          resolve(users);
        }
      })
      .catch(err => {
        _// Handle errors from the request or unexpected responses._
        reject(err);
      });
  });
};`